src/txn/txn.c

```diff
@@ -2811,14 +2821,30 @@ __wt_verbose_dump_txn_one(
         break;
     }
 
+    WT_ERR(__wt_scr_alloc(session, 2048, &snapshot_buf));
+    WT_ERR(__wt_buf_fmt(session, snapshot_buf, "%s", "["));
+    for (i = 0; i < txn->snapshot_data.snapshot_count; i++) {
+        if (i == 0)
+            WT_ERR(__wt_snprintf(
+              snapshot_buf_tmp, sizeof(snapshot_buf_tmp), "%lu", txn->snapshot_data.snapshot[i]));
+        else
+            WT_ERR(__wt_snprintf(
+              snapshot_buf_tmp, sizeof(snapshot_buf_tmp), ", %lu", txn->snapshot_data.snapshot[i]));
```

Comment:
These are nearly identical and it took me a while to see the difference. Can this be combined into a single call that conditionally adds in the ", " based on `i` please?
...(... "%s%lu", i == 0 ? "" : ", ", txn->...).